SLHCUpgradeSimulations/Geometry/test/phase2_reco_pixelntuple_cfg.py

```diff
@@ -43,75 +44,106 @@
+
 process.ReadLocalMeasurement = cms.EDAnalyzer("Phase2PixelNtuple",
-   trackProducer = cms.InputTag("generalTracks"),
-   #verbose = cms.untracked.bool(True),
-   #picky = cms.untracked.bool(False),                                             
-   ### for using track hit association
-   associatePixel = cms.bool(True),
-   associateStrip = cms.bool(False),
-   associateRecoTracks = cms.bool(False),
-   ROUList = cms.vstring('TrackerHitsPixelBarrelLowTof',
-                         'TrackerHitsPixelBarrelHighTof',
-                         'TrackerHitsPixelEndcapLowTof',
-                         'TrackerHitsPixelEndcapHighTof'),
-   usePhase2Tracker = cms.bool(True),
-   pixelSimLinkSrc = cms.InputTag("simSiPixelDigis", "Pixel"),
-   phase2TrackerSimLinkSrc = cms.InputTag("simSiPixelDigis", "Tracker")
-)
+                                              trackProducer = cms.InputTag("generalTracks"),
+                                              trajectoryInput = cms.InputTag('TrackRefitter::USER'),
+                                              #verbose = cms.untracked.bool(True),
+                                              #picky = cms.untracked.bool(False),                                             
+                                              ### for using track hit association
+                                              associatePixel = cms.bool(True),
+                                              associateStrip = cms.bool(False),
+                                              associateRecoTracks = cms.bool(False),
+                                              ROUList = cms.vstring('TrackerHitsPixelBarrelLowTof',
+                                                                    'TrackerHitsPixelBarrelHighTof',
+                                                                    'TrackerHitsPixelEndcapLowTof',
+                                                                    'TrackerHitsPixelEndcapHighTof'),
+                                              ttrhBuilder = cms.string("WithTrackAngle"),
+                                              usePhase2Tracker = cms.bool(True),
+                                              pixelSimLinkSrc = cms.InputTag("simSiPixelDigis", "Pixel"),
+                                              phase2TrackerSimLinkSrc = cms.InputTag("simSiPixelDigis", "Tracker")
+                                          )
 
 
 # Additional output definition
 
 # Other statements
-process.mix.digitizers = cms.PSet(process.theDigitizersValid)
+process.mix.playback = True
+process.mix.digitizers = cms.PSet()
+for a in process.aliases: delattr(process, a)
+process.RandomNumberGeneratorService.restoreStateLabel=cms.untracked.string("randomEngineStateProducer")
 from Configuration.AlCa.GlobalTag import GlobalTag
-process.GlobalTag = GlobalTag(process.GlobalTag, 'auto:phase2_realistic_T15', '')
+process.GlobalTag = GlobalTag(process.GlobalTag, 'auto:phase2_realistic_T21', '')
+
+# CPE generic
```

Comment:
delete any commented-out code in this file that is no longer needed

Reply:
Given that this config is most an example for the phase2 ntuplizer I thought it would be good to leave these in as they are useful options for phase 2 studies. I tried to update the comment of what they refer to. I can remove them if you want though

Reply:
Thanks; it's the right approach to add descriptive comments/notes for commented-out lines of code that should be kept around.